src/sentry/replays/usecases/ingest/event_logger.py

```diff
@@ -163,6 +168,65 @@ def log_option_events(event_meta: ParsedEventMeta, project_id: int, replay_id: s
         logger.info("sentry.replays.slow_click", extra=log)
 
 
+@sentry_sdk.trace
+def log_multiclick_events(event_meta: ParsedEventMeta, project_id: int, replay_id: str) -> None:
+    for multiclick in event_meta.multiclick_events:
+        # Sample multiclick events at 0.2% rate
+        if random.randint(0, 499) >= 1:
+            continue
+
+        is_rage_multiclick = multiclick.click_count >= RAGE_CLICK_COUNT_THRESHOLD
+        log = {
+            "project_id": project_id,
+            "replay_id": replay_id,
+            "alt": multiclick.alt,
+            "aria_label": multiclick.aria_label,
+            "classes": multiclick.classes,
+            "component_name": multiclick.component_name,
+            "id": multiclick.id,
+            "node_id": multiclick.node_id,
+            "role": multiclick.role,
+            "selector": multiclick.selector,
+            "tag": multiclick.tag,
+            "testid": multiclick.testid,
+            "text": multiclick.text[:100],  # Truncate text for logging
+            "timestamp": multiclick.timestamp,
+            "url": multiclick.url or "",
+            "title": multiclick.title,
+            "click_count": multiclick.click_count,
+            "is_rage_multiclick": is_rage_multiclick,
+        }
+        logger.info("sentry.replays.multi_click", extra=log)
+
+
+@sentry_sdk.trace
+def log_rage_click_events(event_meta: ParsedEventMeta, project_id: int, replay_id: str) -> None:
+    for click in event_meta.click_events:
+        # Sample rage click events at 0.2% rate
+        if click.is_rage and random.randint(0, 499) < 1:
```

Comment:
Random makes testing difficult. You can pass a callback as a function argument with `random.randint` as the default. In your test case you can specify anything for this callback and assert functionality works for both states.